src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -0,0 +1,266 @@
+            df[models.MYOCARDIAL_INFARCTION_MODEL_NAME]
+            == models.ACUTE_MYOCARDIAL_INFARCTION_STATE_NAME
+        )
+        emergency_not_already_scheduled = index[
+            (mask_acute_is | mask_acute_mi) & ~(mask_chronic_is | mask_post_mi)
+        ]
+        visitors[data_values.VISIT_TYPES.EMERGENCY] = emergency_not_already_scheduled
+        df.loc[emergency_not_already_scheduled] = self.visit_doctor(
+            df=df.loc[emergency_not_already_scheduled],
+            visitors=visitors,
+            event_time=event_time,
+        )
+
+        self.population_view.update(
+            df[
+                [
+                    self.visit_type_column,
+                    self.scheduled_visit_date_column,
+                    self.miss_scheduled_visit_probability_column,
+                ]
+            ]
+        )
+
+    def on_time_step_cleanup(self, event: Event) -> None:
+        """Determine if someone will go for an emergency visit, background visit,
+        or followup visit. In the event that a simulant goes for an emergency or
+        background visit but already has a followup visit scheduled for the future,
+        keep that scheduled followup and do not schedule another one.
+        """
+        df = self.population_view.get(event.index, query='alive == "alive"')
+        visitors = {}
+
+        # Emergency visits
+        mask_acute_is = (
+            df[models.ISCHEMIC_STROKE_MODEL_NAME] == models.ACUTE_ISCHEMIC_STROKE_STATE_NAME
+        )
+        mask_acute_mi = (
+            df[models.MYOCARDIAL_INFARCTION_MODEL_NAME]
+            == models.ACUTE_MYOCARDIAL_INFARCTION_STATE_NAME
+        )
+        mask_emergency = mask_acute_is | mask_acute_mi
+        visit_emergency = df[mask_emergency].index
+        visitors[data_values.VISIT_TYPES.EMERGENCY] = visit_emergency
+
+        # Scheduled visits
+        mask_scheduled_non_emergency = (
+            (df[self.scheduled_visit_date_column] > (event.time - event.step_size))
+            & (df[self.scheduled_visit_date_column] <= event.time)
+            & (~mask_emergency)
+        )
+        scheduled_non_emergency = df[mask_scheduled_non_emergency].index
+        # Missed scheduled (non-emergency) visits (these do not get re-scheduled)
+        missed_visit = self.randomness.filter_for_probability(
+            scheduled_non_emergency,
+            df.loc[scheduled_non_emergency, self.miss_scheduled_visit_probability_column],
+        )  # pd.Index
+        visitors[data_values.VISIT_TYPES.MISSED] = missed_visit
+        df.loc[missed_visit, self.scheduled_visit_date_column] = pd.NaT  # no re-schedule
+        visit_scheduled = scheduled_non_emergency.difference(missed_visit)
+        visitors[data_values.VISIT_TYPES.SCHEDULED] = visit_scheduled
+
+        # Background visits (for those who did not go for another reason (even missed screenings))
+        maybe_background = df.index.difference(visit_emergency.union(visit_scheduled))
```

Comment:
this looks like everyone who didnt either have an emergency visit or a scheduled visit. 
shouldn't it include anyone who missed their scheduled visit? am i misreading?

Reply:
It does! If you look a couple of lines above, `visit_scheduled = scheduled_non_emergency.difference(missed_visit)`

Reply:
That is, visit_scheduled does not include people who missed their visit by definition